Packs/MicrosoftExchangeOnline/Integrations/EWSO365/EWSO365.py

```diff
@@ -2049,16 +2049,18 @@ def handle_attached_email_with_incorrect_id(attached_email: Message):
     Returns:
         Message: attached email object.
     """
+    demisto.debug('XSUP-32660: handle_attached_email_with_incorrect_message_id running')
     message_id_value = ""
     for i in range(len(attached_email._headers)):
         if attached_email._headers[i][0] == "Message-ID":
             message_id = attached_email._headers[i][1]
+            demisto.debug(f'XSUP-32660: {message_id=}')
             try:
-                if message_id.endswith("]>") and message_id.startswith("<["):
+                if message_id.find("<[") != -1 or message_id.find("]>") != -1:
                     demisto.debug(f"Fixing invalid {message_id=} attachment header by removing its square bracket \
                         wrapper (see XSUP-32074 for further information)")
                     attached_email._headers.pop(i)
-                    message_id_value = f"<{message_id[2:-2]}>"
+                    message_id_value = message_id.replace("<[", "<").replace("]>", ">")
```

Comment:
This may replace unrelated `<[`s (that are not wrappers), try to [use re for replacement](https://www.phind.com/search?q=regex+to+replace+%3C%5Bfoo%5D%3E+with+%3Cfoo%3E%2C+python&ignoreSearchResults=false)

Reply:
Done.